src/overlay/Overlay_D3D12.cpp

```diff
@@ -29,82 +28,182 @@ BOOL CALLBACK EnumWindowsProcMy(HWND hwnd, LPARAM lParam)
     return TRUE;
 }
 
-void Overlay::InitializeD3D12(IDXGISwapChain3* pSwapChain)
+bool Overlay::InitializeD3D12(IDXGISwapChain3* pSwapChain)
 {
-    EnumWindows(EnumWindowsProcMy, reinterpret_cast<LPARAM>(&m_hwnd));
+    static auto checkCmdQueue = [](Overlay* overlay) {
```

Comment:
line skip

Reply:
also maybe it should be its own function and not a lambda?

Reply:
Don't know honestly if I would make these regular functions. They have meaning only in context of this function, both.

But if you think it would be better, no problem to move both outside. But I really think it is unnecessary.
I also tried to made this lambdas in function format so no functors are in play.

Reply:
for now, corrected line skip here...

gonna leave it as lambda for now till you respond